lib/fetch.js

```diff
@@ -0,0 +1,29 @@
+const clownface = require('clownface-io')
+const { contentType } = require('mime-types')
+const { PassThrough } = require('stream')
+const { readable } = require('duplex-to')
+const ns = require('@tpluscode/rdf-ns-builders')
+const { fetch } = require('./protoFetch')
+
+const json = /json$/i
+
+function fetchCsv ({ csvw }) {
+  const csvStream = new PassThrough()
+
+  Promise.resolve().then(async () => {
+    const mappings = await clownface().namedNode(csvw).fetch({
+      contentTypeLookup: (ext) => {
+        return json.test(ext) ? 'application/ld+json' : contentType(ext)
```

Comment:
@bergos I would change clownface-io so that this becomes an optional extension, with `mime-type` as fallback if `undefined` is returned

Reply:
You mean in general detect the media type based on the file extension, if it's a file URL? OK for me.

Reply:
That is what that library does, right? By the way, what does it return for non-file URIs?

Reply:
Ah no, it only gets the extension. And returns `false` for stuff it does no recognize. Curious